server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -219,10 +219,6 @@ protected static boolean parseObjectOrDocumentTypeProperties(String fieldName, O
                 if (value.equalsIgnoreCase("strict")) {
                     builder.dynamic(Dynamic.STRICT);
                 }  else if (value.equalsIgnoreCase("runtime")) {
-                    if (parserContext.supportsDynamicRuntimeMappings() == false) {
```

Comment:
So good to see this gone!

Reply:
yes that is what I thought too. There is more to do to simplify the abstractions here, but removing the extension point and some of the shenanigans around it feels good.